### ISSUE TITLE
Correctly set `IsRecording` from default OTel default global impl

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go-instrumentation#787]
To unblock the beta we decided to return `true` by default and just respond incorrectly when the span would have been dropped.

Before GA we should try to actually returning the correct value.

_Originally posted by @MrAlias in https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/537#issuecomment-2059576498_

**Comment by MrAlias** [comment]
Closing in favor of https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/954
